### PR TITLE
Fixes 3790: Adding functionality to support isolate_namespace engines.

### PR DESCRIPTION
**Pull request by ehelms**
When engines are defined with isolate_namespace, their routes and
route resolutions are not directly available to the parent application.
Thus, this adds the ability to specify a path directly by the plugin
for a given menu item. Further, since isolated engines may include
entire menus of their own and wish to separate the items, a divider
function is added.

### CODE REVIEW

**Comment by ohadlevy** [comment]
@abenari can you review please?

**Comment by ehelms** [comment]
This was inspired by initial work being done in Katello to support a single navigation. How the plugin is defined using the updates here can be seen here - https://github.com/Katello/katello/pull/3446

File: app/services/menu/manager.rb [hunk 1]
```diff
@@ -30,7 +30,7 @@ def map(menu_name)
 
       def items(menu_name)
         # force menu reload in development when auto loading modified files
-        @items || Menu::Loader.load
+        @items || {}
```
Comment (ohadlevy) [inline app/services/menu/manager.rb:33]: Why did you remove this? did it cause any issues?

Comment (ehelms) [inline app/services/menu/manager.rb:33]: What I was seeing was that in development mode, the menu items would get completely reloaded and all of what was added by the plugin on initial boot of Foreman would be lost from that point on.

**Comment by ehelms** [comment]
Closing in favor of https://github.com/theforeman/foreman/pull/1083 for divider functionality and https://github.com/theforeman/foreman/pull/1082
